gradle-plugin/src/proguard/gradle/ProGuardTask.java

```diff
@@ -1384,17 +1379,37 @@ public void proguard()
         loggingManager.captureStandardError(LogLevel.WARN);
 
         // Run ProGuard with the collected configuration.
-        new ProGuard(getConfiguration()).execute();
+        new ProGuard(getConfiguration(getConfigurationFiles().getElements().get())).execute();
```

Comment:
Why don't you query the `configurationProvider` here as well?

Reply:
Good catch. Fixed in 8882e7a